Feature/optimize docker script and update docs (#1190)
apache/karaf: assemblies/docker/README.md

## Build

Images are based on the Docker official [AdoptOpenJDK 11 JRE Hotspot](https://hub.docker.com/_/adoptopenjdk?tab=tags&page=1&name=11-jre-hotspot) image. If you want to
build the Karaf image you have the following choices:

> 1. Create the docker image from the SNAPSHOT version (from local distribution)

Reviewer: That's not accurate: local distribution is not necessary SNAPSHOT. It could be a custom distribution locally built (like on CI). I would change this with:

`1. Create the docker image from a local distribution package`

2. Create the docker image from an Apache Karaf archive, for example (apache-karaf-4.2.9.tar.gz)

> 3. Create the docker image from a specific version of Apache Karaf

Reviewer: I would add remote or local custom Karaf distribution.

If you run `build.sh` without arguments then you could see how to usage this command.

```bash
Usage:
  build.sh --from-local-dist [--archive <archive>] [--image-name <image>] [--build-multi-platform <comma-separated platforms>]
  build.sh --from-release --karaf-version <x.x.x> [--image-name <image>] [--build-multi-platform <comma-separated platforms>]
  build.sh --help

  If the --image-name flag is not used the built image name will be 'karaf'.
  Check the supported build platforms; you can verify with this command: docker buildx ls
  The supported platforms (OS/Arch) depend on the build's base image, in this case [adoptopenjdk:11-jre-hotspot](https://hub.docker.com/_/adoptopenjdk?tab=tags&page=1&name=11-jre-hotspot).
```

> To create the docker image from the SNAPSHOT version (from local distribution) 

Reviewer: I would remove this paragraph. It could be confusing for users.

> For create the docker image from the SNAPSHOT version (from local distribution) 
> you can execute the below command. Remember that before you can successfully run 
> this command, you must have done the build of the project 

Reviewer 2: ```suggestion
To create the docker image from the SNAPSHOT version (from local distribution) 
you can execute the command below. Remember that before you can successfully run 
this command, you must build the project 
```

(for example with the command `mvn clean install -DskipTests`). For more info
you can read: [Building Apache Karaf](https://github.com/apache/karaf/blob/master/BUILDING.md#building-apache-karaf)

```bash
./build.sh --from-local-dist
```

> For create the docker image from the local dist version but with the archive,
> you can execute the below command. Remember that before you can successfully run 
> this command.

Reviewer 2: ```suggestion
To create the docker image from the local dist version but with the archive,
you can execute the command below. Remember that before you can successfully run 
this command, you must build the project.
```

```bash
./build.sh --from-local-dist --archive ~/home/amusarra/apache-karaf-4.2.9.tar.gz
```

> You can also specify the image name with the flag `--image-name`, for example

Reviewer 2: ```suggestion
You can also specify the image name with the `--image-name` flag, for example
```

(replacing the version, image name, and targets as appropriate):

```bash

> ./build.sh --from-local-dist --archive ~/Downloads/apache-karaf-4.2.9.tar.gz --image-name amusarra/karaf:4.2.9-dev

Reviewer: Please don't use user specific setup. I would change to something like:

`--image-name myrepo/mykaraf:x.x.x`

to be generic.

```

If you want to build the docker image for a specific version of Karaf
you can run `build.sh` command in this way (replacing the version, image name, 
and targets as appropriate):

```bash

> ./build.sh --from-release --karaf-version 4.2.9 --image-name amusarra/karaf:4.2.9

Reviewer: Same here: please avoid specific user content (basically replace amusarra with myrepo for instance).
